### ISSUE TITLE
language service warns about ES private field emit in declaration files (`#private`)

### ISSUE THREAD

**Issue by mheiber** [microsoft/TypeScript#36963]
**TypeScript Version:** 3.9.0-dev.20200222

**Search Terms:**

private fields, es private fields, declaration files

**Code**

```ts
export class A {
    #foo = 3;
    constructor() {
        console.log(this.#foo);
    }
}
```

**Expected behavior:**

`npx typescript@next 1.ts --target esnext --declaration` should produce a declaration file that the language service doesn't complain about.

**Actual behavior:**

```ts
export declare class A {
    #private; // two language service warnings here
    constructor();
}
```

The language service complains about the following issues in the declaration file:

> Member '#private' implicitly has 'any' type, but a better type may be inferred from usage. ts(7045)
> '#private' is declared but its value is never read. ts(6133)'

**Playground Link:** 

Can't do, since the playground doesn't yet support dts generation

**Suggested solution**

To get rid of the warning about implicit any, we could change the emit to:

```ts
export declare class A {
    #private: unknown;
    constructor();
}
```

To get rid of the warning about an unused variable, we could update the language service to never warn on unused private fields in *declarations*. I think this rule would ideally be applied for declarations anywhere (not just in declaration files), since the use cases for forcing nominality are probably the same in .ts files.

Happy to implement!

**Comment by sandersn** [comment]
@mheiber I think in your first example the emit should not include `unknown`, right?

```ts
export declare class A {
    #private; // two language service warnings here
    constructor();
}
```

That's what I get when trying to repro this, at least.

**Comment by sandersn** [comment]
The implicit any error was fixed in https://github.com/microsoft/TypeScript/pull/36640. I note that the language service puts this warning on lots of different ambient declarations:

```ts
export declare class AA {
    #private; // warning
    private y; // warning
    x: number; // ok
    constructor();
}
declare class BB { // warning
    x: number // ok
}
```

I don't think this is worth a patch fix since it's just a language service warning, and one people have been living with on `private` since it was introduced, I assume. It's also not clear whether the `BB` warning should be removed either, so there might be a little design thought needed to fix this bug.

(Also there is an almost-done version of the playground that does have dts output: https://www.typescriptlang.org/v2/en/play. I'm not sure if the url reflects which output tab is selected. However, the playground doesn't request or show suggestion diagnostics, so it's still not usable to repro this.)

**Comment by mheiber** [comment]
@

> 
> 
> @mheiber I think in your first example the emit should not include `unknown`, right?
> 
> ```ts
> export declare class A {
>     #private; // two language service warnings here
>     constructor();
> }
> ```
> 
> That's what I get when trying to repro this, at least.

That's right, I pasted wrong. Thanks for checking.

**Comment by mheiber** [comment]
Thanks @sandersn !
